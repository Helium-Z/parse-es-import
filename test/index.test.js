import fs from 'fs';
import path from 'path';
import parse from '../es';

test('it matches the snapshot for each app example fixture', () => {
  const fixturesPath = path.resolve(__dirname, './fixtures');
  const examplePaths = fs
    .readdirSync(fixturesPath)
    .map((filename) => path.join(fixturesPath, filename));

  examplePaths.forEach((demoPath) => {
    const demoContent = fs.readFileSync(demoPath, 'utf8');
    const result = {
      result: parse(demoContent),
      name: path.basename(demoPath),
    };

    expect(result).toMatchSnapshot();
  });
});
