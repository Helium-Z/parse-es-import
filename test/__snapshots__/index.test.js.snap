// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot for each app example fixture 1`] = `
Object {
  "name": "demo-1.jsx",
  "result": Object {
    "exports": Array [
      Object {
        "endIndex": 198,
        "moduleName": "Demo",
        "startIndex": 136,
        "type": "FunctionDeclaration",
        "value": "function Demo() {
  return <div>Hello World...</div>;
}",
      },
    ],
    "imports": Array [
      Object {
        "defaultImport": "React",
        "endIndex": 26,
        "moduleName": "react",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 0,
      },
      Object {
        "defaultImport": "antd",
        "endIndex": 83,
        "moduleName": "antd",
        "namedImports": Array [
          Object {
            "alias": "AntButton",
            "name": "Button",
          },
          Object {
            "alias": "Alert",
            "name": "Alert",
          },
        ],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 27,
      },
      Object {
        "defaultImport": "",
        "endIndex": 115,
        "moduleName": "hello",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "Hello",
        "startIndex": 84,
      },
      Object {
        "defaultImport": "",
        "endIndex": 134,
        "moduleName": "xx.less",
        "namedImports": Array [],
        "sideEffectOnly": true,
        "starImport": "",
        "startIndex": 117,
      },
    ],
  },
}
`;

exports[`it matches the snapshot for each app example fixture 2`] = `
Object {
  "name": "demo-2.jsx",
  "result": Object {
    "exports": Array [],
    "imports": Array [
      Object {
        "defaultImport": "React",
        "endIndex": 156,
        "moduleName": "react",
        "namedImports": Array [
          Object {
            "alias": "foo",
            "name": "useState",
          },
          Object {
            "alias": "useEffect",
            "name": "useEffect",
          },
          Object {
            "alias": "useLayoutEffect",
            "name": "useLayoutEffect",
          },
          Object {
            "alias": "useRef",
            "name": "useRef",
          },
          Object {
            "alias": "useReducer",
            "name": "useReducer",
          },
        ],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 0,
      },
      Object {
        "defaultImport": "",
        "endIndex": 192,
        "moduleName": "react-dom",
        "namedImports": Array [
          Object {
            "alias": "render",
            "name": "render",
          },
        ],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 157,
      },
      Object {
        "defaultImport": "",
        "endIndex": 229,
        "moduleName": "./logo.svg",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "logos",
        "startIndex": 193,
      },
      Object {
        "defaultImport": "",
        "endIndex": 249,
        "moduleName": "./App.css",
        "namedImports": Array [],
        "sideEffectOnly": true,
        "starImport": "",
        "startIndex": 230,
      },
    ],
  },
}
`;

exports[`it matches the snapshot for each app example fixture 3`] = `
Object {
  "name": "demo-3.jsx",
  "result": Object {
    "exports": Array [],
    "imports": Array [
      Object {
        "defaultImport": "React",
        "endIndex": 26,
        "moduleName": "react",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 0,
      },
      Object {
        "defaultImport": "",
        "endIndex": 62,
        "moduleName": "react-dom",
        "namedImports": Array [
          Object {
            "alias": "render",
            "name": "render",
          },
        ],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 27,
      },
      Object {
        "defaultImport": "",
        "endIndex": 84,
        "moduleName": "./index.css",
        "namedImports": Array [],
        "sideEffectOnly": true,
        "starImport": "",
        "startIndex": 63,
      },
      Object {
        "defaultImport": "App",
        "endIndex": 109,
        "moduleName": "./App",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "",
        "startIndex": 85,
      },
      Object {
        "defaultImport": "",
        "endIndex": 159,
        "moduleName": "./serviceWorker",
        "namedImports": Array [],
        "sideEffectOnly": false,
        "starImport": "serviceWorker",
        "startIndex": 110,
      },
      Object {
        "defaultImport": "",
        "endIndex": 1179,
        "moduleName": "./really/long/import/ea/ut/exercitation/commodo/ad/ea/reprehenderit/nisi/deserunt/sint/ipsum/consectetur/occaecat/dolor/minim/dolor/consectetur/ad/reprehenderit/aute/nisi/duis/labore/eu/reprehenderit/deserunt/incididunt/quis/deserunt/ex/incididunt/occaecat/magna/veniam/sunt/anim/id/qui/non/consectetur/est/ad/laboris/velit/non/irure/cillum/sint/consectetur/cillum/deserunt/id/do/dolor/labore/pariatur/qui/est/qui/incididunt/dolore/eu/cupidatat/ea/pariatur/aliqua/laboris/excepteur/eiusmod/culpa/voluptate/ullamco/sint/dolor/do/incididunt/aliquip/irure/id/in/occaecat/duis/id/et/magna/id/enim/officia/aliquip/exercitation/cupidatat/quis/ipsum/non/incididunt/adipisicing/voluptate/quis/nostrud/veniam/magna/esse/excepteur/officia/lorem/ipsum/officia/fugiat/proident/amet/culpa/esse/amet/sit/esse/dolor/commodo/voluptate/excepteur/labore/cupidatat/ad/eu/ad/dolor/id/aliqua/in/veniam/laboris/excepteur/exercitation/officia/aliquip/elit/aliquip/dolore/sit/fugiat/fugiat/officia/fugiat/commodo/amet/laboris/nostrud",
        "namedImports": Array [],
        "sideEffectOnly": true,
        "starImport": "",
        "startIndex": 160,
      },
    ],
  },
}
`;
